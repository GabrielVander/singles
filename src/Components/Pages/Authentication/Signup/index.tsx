import React, {useState} from "react";
import {message} from "antd";
import './styles.css';
import {Credentials, useFirebase} from "react-redux-firebase";
import {useHistory} from "react-router-dom";
import {HOME} from "../../../../Routes/AppRoutes";
import AuthMethod from "../../../../Model/Authentication/AuthMethod";
import {User, UserInfo} from "firebase";
import CenteredSpin from "../../../Others/CenteredSpin";
import {useTranslation} from "react-i18next";
import SignupForm from "./SignupForm";
import AuthLayout from "../../../Others/AuthLayout";

function Signup() {
    const firebase = useFirebase();
    const history = useHistory();
    const {t} = useTranslation(['login']);

    const [loading, setLoading] = useState<boolean>(false);

    function signInWithEmailAndPassword(email: string, password: string) {
        setLoading(true);

        firebase.createUser({
            email,
            password,
        })
            .then(result => {
                message.success(`Successfully logged in as ${result!.displayName || result!.email}`);
                history.push(HOME.path);
            })
            .catch(reason => handleAuthError(reason));
    }

    function signInWithGoogle() {
        signIn(AuthMethod.GOOGLE)
    }

    function signInWithFacebook() {
        signIn(AuthMethod.FACEBOOK)
    }

    function signInWithTwitter() {
        signIn(AuthMethod.TWITTER)
    }

    function signIn(signInMethod: () => Credentials): void {
        setLoading(true);

        firebase.login(signInMethod())
            .then(result => handleAuthSuccess(result))
            .catch(reason => handleAuthError(reason));
    }

    function handleAuthSuccess(result: { user: User | UserInfo | null, displayName?: string }): void {
        const user = result.user;

        message.success(`Successfully logged in as ${user!.displayName || user!.email}`);
        history.push(HOME.path);
    }

    function handleAuthError(error: { code: string; message: string; }): void {
        message.error(`${(error.code)} - ${(error.message)}`, 5);
        setLoading(false);
    }

    if (loading) {
        return <CenteredSpin tip={t('login:loggingIn')}/>
    }

    return (
        <AuthLayout>
            <SignupForm
                signInWithEmailAndPassword={signInWithEmailAndPassword}
                signInWithGoogle={signInWithGoogle}
                signInWithFacebook={signInWithFacebook}
                signInWithTwitter={signInWithTwitter}/>
        </AuthLayout>
    );
}

export default Signup;
