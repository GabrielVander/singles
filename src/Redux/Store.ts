import {createStore} from "redux";
import rootReducer from "./Reducers";

class Store {
    static INITIAL_GLOBAL_STATE = {};
    // @ts-ignore
    static store = createStore(rootReducer, Store.INITIAL_GLOBAL_STATE, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
}

export default Store;
