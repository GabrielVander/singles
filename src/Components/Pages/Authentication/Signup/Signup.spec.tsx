import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import Signup from "./index";
import Store from "../../../../Redux/Store";

test('matches snapshot', () => {
    const component = render(
        <Provider store={Store.store}>
            <BrowserRouter>
                <Signup/>
            </BrowserRouter>
        </Provider>
    );
    expect(component).toMatchSnapshot();
})
