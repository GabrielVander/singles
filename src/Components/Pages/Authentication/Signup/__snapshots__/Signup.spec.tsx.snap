// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="ant-layout"
        style="height: 100%;"
      >
        <main
          class="ant-layout-content"
        >
          <div
            class="sc-bdnylx kMwTOS"
          >
            <div
              class="ant-row ant-row-space-around ant-row-middle main-row"
            >
              <div
                class="ant-col ant-col-xs-16 ant-col-sm-16 ant-col-md-16 ant-col-lg-12 ant-col-xl-12"
              >
                <div
                  class="sc-bdnylx kMwTOS"
                >
                  <img
                    alt="logo"
                    class="logo"
                    src="logo.svg"
                  />
                </div>
              </div>
              <div
                class="ant-col ant-col-xs-16 ant-col-sm-16 ant-col-md-16 ant-col-lg-12 ant-col-xl-12"
              >
                <form
                  class="ant-form ant-form-horizontal"
                  id="register"
                >
                  <div
                    class="ant-row ant-row-space-around ant-row-middle"
                  >
                    <div
                      class="ant-col"
                    >
                      <h1
                        class="signup-title"
                      >
                        Signup
                      </h1>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-row-middle"
                  >
                    <div
                      class="ant-col"
                      style="flex-basis: auto;"
                    >
                      <div
                        class="ant-row ant-form-item"
                      >
                        <div
                          class="ant-col ant-form-item-label ant-form-item-label-left"
                        >
                          <label
                            class="ant-form-item-required"
                            for="register_email"
                            title="E-mail"
                          >
                            E-mail
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <input
                                class="ant-input"
                                id="register_email"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-row-middle"
                  >
                    <div
                      class="ant-col"
                      style="flex-basis: auto;"
                    >
                      <div
                        class="ant-row ant-form-item"
                      >
                        <div
                          class="ant-col ant-form-item-label ant-form-item-label-left"
                        >
                          <label
                            class="ant-form-item-required"
                            for="register_password"
                            title="Password"
                          >
                            Password
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <span
                                class="ant-input-affix-wrapper ant-input-password"
                              >
                                <input
                                  action="click"
                                  class="ant-input"
                                  id="register_password"
                                  type="password"
                                  value=""
                                />
                                <span
                                  class="ant-input-suffix"
                                >
                                  <span
                                    aria-label="eye-invisible"
                                    class="anticon anticon-eye-invisible ant-input-password-icon"
                                    role="img"
                                    tabindex="-1"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class=""
                                      data-icon="eye-invisible"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                      />
                                      <path
                                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-row-middle"
                  >
                    <div
                      class="ant-col"
                      style="flex-basis: auto;"
                    >
                      <div
                        class="ant-row ant-form-item"
                      >
                        <div
                          class="ant-col ant-form-item-label ant-form-item-label-left"
                        >
                          <label
                            class="ant-form-item-required"
                            for="register_confirm"
                            title="Confirm Password"
                          >
                            Confirm Password
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <span
                                class="ant-input-affix-wrapper ant-input-password"
                              >
                                <input
                                  action="click"
                                  class="ant-input"
                                  id="register_confirm"
                                  type="password"
                                  value=""
                                />
                                <span
                                  class="ant-input-suffix"
                                >
                                  <span
                                    aria-label="eye-invisible"
                                    class="anticon anticon-eye-invisible ant-input-password-icon"
                                    role="img"
                                    tabindex="-1"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class=""
                                      data-icon="eye-invisible"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                      />
                                      <path
                                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-row-space-around ant-row-middle"
                  >
                    <div
                      class="ant-col"
                    >
                      <div
                        class="ant-row ant-form-item"
                      >
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <button
                                class="ant-btn ant-btn-primary"
                                type="submit"
                              >
                                <span>
                                  Register
                                </span>
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-plain"
                    role="separator"
                  >
                    <span
                      class="ant-divider-inner-text"
                    >
                      loginForm:oauthSubdivision
                    </span>
                  </div>
                  <div
                    class="ant-row ant-row-space-around ant-row-middle"
                    style="margin: -12px -8px 12px -8px;"
                  >
                    <div
                      class="ant-col oauth-button ant-col-xs-24 ant-col-xl-8"
                      style="padding: 12px 8px 12px 8px;"
                    >
                      <div
                        class="sc-bdnylx kMwTOS"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span
                            aria-label="google-circle"
                            class="anticon anticon-google-circle"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="google-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm167 633.6C638.4 735 583 757 516.9 757c-95.7 0-178.5-54.9-218.8-134.9C281.5 589 272 551.6 272 512s9.5-77 26.1-110.1c40.3-80.1 123.1-135 218.8-135 66 0 121.4 24.3 163.9 63.8L610.6 401c-25.4-24.3-57.7-36.6-93.6-36.6-63.8 0-117.8 43.1-137.1 101-4.9 14.7-7.7 30.4-7.7 46.6s2.8 31.9 7.7 46.6c19.3 57.9 73.3 101 137 101 33 0 61-8.7 82.9-23.4 26-17.4 43.2-43.3 48.9-74H516.9v-94.8h230.7c2.9 16.1 4.4 32.8 4.4 50.1 0 74.7-26.7 137.4-73 180.1z"
                              />
                            </svg>
                          </span>
                          <span>
                            Google
                          </span>
                        </button>
                      </div>
                    </div>
                    <div
                      class="ant-col oauth-button ant-col-xs-24 ant-col-xl-8"
                      style="padding: 12px 8px 12px 8px;"
                    >
                      <div
                        class="sc-bdnylx kMwTOS"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span
                            aria-label="facebook"
                            class="anticon anticon-facebook"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="facebook"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                              />
                            </svg>
                          </span>
                          <span>
                            Facebook
                          </span>
                        </button>
                      </div>
                    </div>
                    <div
                      class="ant-col oauth-button ant-col-xs-24 ant-col-xl-8"
                      style="padding: 12px 8px 12px 8px;"
                    >
                      <div
                        class="sc-bdnylx kMwTOS"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="button"
                        >
                          <span
                            aria-label="twitter-circle"
                            class="anticon anticon-twitter-circle"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="twitter-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 01-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 01-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 00229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                              />
                            </svg>
                          </span>
                          <span>
                            Twitter
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </main>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="ant-layout"
      style="height: 100%;"
    >
      <main
        class="ant-layout-content"
      >
        <div
          class="sc-bdnylx kMwTOS"
        >
          <div
            class="ant-row ant-row-space-around ant-row-middle main-row"
          >
            <div
              class="ant-col ant-col-xs-16 ant-col-sm-16 ant-col-md-16 ant-col-lg-12 ant-col-xl-12"
            >
              <div
                class="sc-bdnylx kMwTOS"
              >
                <img
                  alt="logo"
                  class="logo"
                  src="logo.svg"
                />
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-16 ant-col-sm-16 ant-col-md-16 ant-col-lg-12 ant-col-xl-12"
            >
              <form
                class="ant-form ant-form-horizontal"
                id="register"
              >
                <div
                  class="ant-row ant-row-space-around ant-row-middle"
                >
                  <div
                    class="ant-col"
                  >
                    <h1
                      class="signup-title"
                    >
                      Signup
                    </h1>
                  </div>
                </div>
                <div
                  class="ant-row ant-row-middle"
                >
                  <div
                    class="ant-col"
                    style="flex-basis: auto;"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label ant-form-item-label-left"
                      >
                        <label
                          class="ant-form-item-required"
                          for="register_email"
                          title="E-mail"
                        >
                          E-mail
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input"
                              id="register_email"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-row-middle"
                >
                  <div
                    class="ant-col"
                    style="flex-basis: auto;"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label ant-form-item-label-left"
                      >
                        <label
                          class="ant-form-item-required"
                          for="register_password"
                          title="Password"
                        >
                          Password
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class="ant-input-affix-wrapper ant-input-password"
                            >
                              <input
                                action="click"
                                class="ant-input"
                                id="register_password"
                                type="password"
                                value=""
                              />
                              <span
                                class="ant-input-suffix"
                              >
                                <span
                                  aria-label="eye-invisible"
                                  class="anticon anticon-eye-invisible ant-input-password-icon"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class=""
                                    data-icon="eye-invisible"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                    />
                                    <path
                                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-row-middle"
                >
                  <div
                    class="ant-col"
                    style="flex-basis: auto;"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label ant-form-item-label-left"
                      >
                        <label
                          class="ant-form-item-required"
                          for="register_confirm"
                          title="Confirm Password"
                        >
                          Confirm Password
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class="ant-input-affix-wrapper ant-input-password"
                            >
                              <input
                                action="click"
                                class="ant-input"
                                id="register_confirm"
                                type="password"
                                value=""
                              />
                              <span
                                class="ant-input-suffix"
                              >
                                <span
                                  aria-label="eye-invisible"
                                  class="anticon anticon-eye-invisible ant-input-password-icon"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class=""
                                    data-icon="eye-invisible"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                    />
                                    <path
                                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-row-space-around ant-row-middle"
                >
                  <div
                    class="ant-col"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <button
                              class="ant-btn ant-btn-primary"
                              type="submit"
                            >
                              <span>
                                Register
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-plain"
                  role="separator"
                >
                  <span
                    class="ant-divider-inner-text"
                  >
                    loginForm:oauthSubdivision
                  </span>
                </div>
                <div
                  class="ant-row ant-row-space-around ant-row-middle"
                  style="margin: -12px -8px 12px -8px;"
                >
                  <div
                    class="ant-col oauth-button ant-col-xs-24 ant-col-xl-8"
                    style="padding: 12px 8px 12px 8px;"
                  >
                    <div
                      class="sc-bdnylx kMwTOS"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="button"
                      >
                        <span
                          aria-label="google-circle"
                          class="anticon anticon-google-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="google-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm167 633.6C638.4 735 583 757 516.9 757c-95.7 0-178.5-54.9-218.8-134.9C281.5 589 272 551.6 272 512s9.5-77 26.1-110.1c40.3-80.1 123.1-135 218.8-135 66 0 121.4 24.3 163.9 63.8L610.6 401c-25.4-24.3-57.7-36.6-93.6-36.6-63.8 0-117.8 43.1-137.1 101-4.9 14.7-7.7 30.4-7.7 46.6s2.8 31.9 7.7 46.6c19.3 57.9 73.3 101 137 101 33 0 61-8.7 82.9-23.4 26-17.4 43.2-43.3 48.9-74H516.9v-94.8h230.7c2.9 16.1 4.4 32.8 4.4 50.1 0 74.7-26.7 137.4-73 180.1z"
                            />
                          </svg>
                        </span>
                        <span>
                          Google
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="ant-col oauth-button ant-col-xs-24 ant-col-xl-8"
                    style="padding: 12px 8px 12px 8px;"
                  >
                    <div
                      class="sc-bdnylx kMwTOS"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="button"
                      >
                        <span
                          aria-label="facebook"
                          class="anticon anticon-facebook"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="facebook"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                            />
                          </svg>
                        </span>
                        <span>
                          Facebook
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="ant-col oauth-button ant-col-xs-24 ant-col-xl-8"
                    style="padding: 12px 8px 12px 8px;"
                  >
                    <div
                      class="sc-bdnylx kMwTOS"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="button"
                      >
                        <span
                          aria-label="twitter-circle"
                          class="anticon anticon-twitter-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="twitter-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 01-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 01-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 00229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                            />
                          </svg>
                        </span>
                        <span>
                          Twitter
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
